src/univariate/discrete/dirac.jl

```diff
@@ -0,0 +1,30 @@
+"""
+    Dirac(value)
+
+A *Dirac distribution* is parametrized by its only value, and takes its value with probability 1.
+
+```julia
+d = Dirac(2.)  # Dirac distribution with value = 2.
+rand(d)       # Always returns the same value
+```
+"""
+struct Dirac{T} <: DiscreteUnivariateDistribution
+    value::T
+end
+
+eltype(::Dirac{T}) where {T} = T
+rand(rng::AbstractRNG, d::Dirac) = d.value
+pdf(d::Dirac, x::Real) = x == d.value ? 1 : 0
```

Comment:
these should be floating point
```suggestion
pdf(d::Dirac, x::Real) = x == d.value ? 1.0 : 0.0
```

Reply:
Shouldn't `pdf(d, NaN)` return `NaN`? Or what if `d.value` is `NaN`?